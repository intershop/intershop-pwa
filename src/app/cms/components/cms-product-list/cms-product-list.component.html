<div *ngIf="productSKUs.length" class="product-list-container" [ngClass]="pagelet.stringParam('CSSClass')">
  <h2 *ngIf="pagelet.hasParam('Title')">{{ pagelet.stringParam('Title') }}</h2>
  <div class="product-list row">
    <div *ngFor="let productSKU of productSKUs"
      class="product-list-item"
      [ngClass]="pagelet.stringParam('ListItemCSSClass')"
    >
      <ish-product-tile-container
        [productSku]="productSKU"
      ></ish-product-tile-container>
    </div>
  </div>
</div>

<!-- TODO: (CMS Component Configuration)
  ProductView
  ListStyle
  SlideItems
-->
