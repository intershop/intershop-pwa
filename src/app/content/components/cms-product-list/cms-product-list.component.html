<div *ngIf="getConfigurationParameterValue(pagelet, 'Products') && getConfigurationParameterValue(pagelet, 'Products').length > 0" class="product-list-container" [ngClass]="getConfigurationParameterValue(pagelet, 'CSSClass')">
  <h2 *ngIf="getConfigurationParameterValue(pagelet, 'Title')">{{ getConfigurationParameterValue(pagelet, 'Title') }}</h2>
  <div class="product-list row">
    <div *ngFor="let product of getConfigurationParameterValue(pagelet, 'Products')"
      class="product-list-item"
      [ngClass]="getConfigurationParameterValue(pagelet, 'ListItemCSSClass')"
    >
      <ish-product-tile-container
        [productSku]="product.split('@')[0]"
      ></ish-product-tile-container>
    </div>
  </div>
</div>

<!-- TODO: (CMS Component Configuration)
  ProductView
  ListStyle
  SlideItems
-->
