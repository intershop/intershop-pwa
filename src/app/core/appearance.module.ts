import { NgModule } from '@angular/core';
import { NgbPopoverConfig } from '@ng-bootstrap/ng-bootstrap';
import { ToastrModule } from 'ngx-toastr';

import { IconModule } from './icon.module';
import { ThemeService } from './utils/theme/theme.service';

@NgModule({
  imports: [
    IconModule,
    ToastrModule.forRoot({
      closeButton: true,
      timeOut: 3000,
      positionClass: 'toast-top-full-width', // toast-top-center
      preventDuplicates: true,
    }),
  ],
})
export class AppearanceModule {
  constructor(popoverConfig: NgbPopoverConfig, themeService: ThemeService) {
    popoverConfig.placement = 'top';
    popoverConfig.triggers = 'hover';
    popoverConfig.container = 'body';

    themeService.init();
  }
}
