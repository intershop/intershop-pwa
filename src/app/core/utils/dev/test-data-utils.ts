import { CategoryTree, CategoryTreeHelper } from 'ish-core/models/category-tree/category-tree.model';
import { Category } from 'ish-core/models/category/category.model';
import { ContentPageTreeHelper } from 'ish-core/models/content-page-tree/content-page-tree.helper';
import { ContentPageTree, ContentPageTreeElement } from 'ish-core/models/content-page-tree/content-page-tree.model';

export function categoryTree(categories?: Category[]): CategoryTree {
  const tree = CategoryTreeHelper.empty();
  if (categories) {
    return categories.reduce((acc, cat) => CategoryTreeHelper.add(acc, cat), tree);
  }
  return tree;
}

export function pageTree(elements?: ContentPageTreeElement[]): ContentPageTree {
  const tree = ContentPageTreeHelper.empty();
  if (elements) {
    return elements.reduce((acc, cat) => ContentPageTreeHelper.add(acc, cat), tree);
  }
  return tree;
}
