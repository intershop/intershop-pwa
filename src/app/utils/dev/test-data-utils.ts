import { CategoryTree, CategoryTreeHelper } from 'ish-core/models/category-tree/category-tree.model';
import { Category } from 'ish-core/models/category/category.model';

export function categoryTree(categories?: Category[]): CategoryTree {
  const tree = CategoryTreeHelper.empty();
  if (categories) {
    return categories.reduce((acc, cat) => CategoryTreeHelper.add(acc, cat), tree);
  }
  return tree;
}
