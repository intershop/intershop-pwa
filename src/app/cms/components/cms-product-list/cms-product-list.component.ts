import { ChangeDetectionStrategy, Component, Input, OnChanges } from '@angular/core';

import { ContentPageletView } from 'ish-core/models/content-view/content-views';
import { SfeMetadataWrapper } from '../../../cms/sfe-adapter/sfe-metadata-wrapper';

@Component({
  selector: 'ish-cms-product-list',
  templateUrl: './cms-product-list.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CMSProductListComponent extends SfeMetadataWrapper implements OnChanges {
  @Input() pagelet: ContentPageletView;
  productSKUs: string[] = [];

  ngOnChanges() {
    this.productSKUs = this.pagelet.hasParam('Products')
      ? this.pagelet.configParam<string[]>('Products').map(product => product.split('@')[0])
      : [];
  }
}
