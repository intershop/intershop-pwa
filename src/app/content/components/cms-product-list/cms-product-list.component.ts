import { ChangeDetectionStrategy, Component, DoCheck, Input } from '@angular/core';

import { ContentPageletView } from '../../../models/content-view/content-views';

@Component({
  selector: 'ish-cms-product-list',
  templateUrl: './cms-product-list.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CMSProductListComponent implements DoCheck {
  @Input()
  pagelet: ContentPageletView;

  productSKUs: string[] = [];

  ngDoCheck() {
    if (!this.pagelet || !this.pagelet.hasParam('Products')) {
      this.productSKUs = [];
      return;
    }

    this.productSKUs = this.pagelet.configParam<string[]>('Products').map(product => product.split('@')[0]);
  }
}
