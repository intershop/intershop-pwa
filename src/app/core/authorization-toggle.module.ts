import { ModuleWithProviders, NgModule } from '@angular/core';
import { ReplaySubject } from 'rxjs';
import { map } from 'rxjs/operators';

import { AuthorizationToggleDirective } from './directives/authorization-toggle.directive';
import { AuthorizationToggleService, checkPermission } from './utils/authorization-toggle/authorization-toggle.service';
import { whenTruthy } from './utils/operators';

@NgModule({
  declarations: [AuthorizationToggleDirective],
  exports: [AuthorizationToggleDirective],
})
export class AuthorizationToggleModule {
  private static permissions = new ReplaySubject<string[]>(1);

  static forTesting(...permissions: string[]): ModuleWithProviders<AuthorizationToggleModule> {
    AuthorizationToggleModule.switchTestingPermissions(...permissions);
    return {
      ngModule: AuthorizationToggleModule,
      providers: [
        {
          provide: AuthorizationToggleService,
          useValue: {
            isAuthorizedTo: (permission: string) =>
              AuthorizationToggleModule.permissions.pipe(
                whenTruthy(),
                map(perms => checkPermission(perms, permission))
              ),
          },
        },
      ],
    };
  }

  static switchTestingPermissions(...permissions: string[]) {
    AuthorizationToggleModule.permissions.next(permissions);
  }
}

export { AuthorizationToggleService } from './utils/authorization-toggle/authorization-toggle.service';
export { AuthorizationToggleGuard } from './guards/authorization-toggle.guard';
