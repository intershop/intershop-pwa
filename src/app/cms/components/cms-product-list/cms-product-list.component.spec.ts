import { ComponentFixture, TestBed, async } from '@angular/core/testing';

import { MockComponent } from 'ish-core/utils/dev/mock.component';
import { createSimplePageletView } from 'ish-core/utils/dev/test-data-utils';

import { CMSProductListComponent } from './cms-product-list.component';

describe('Cms Product List Component', () => {
  let component: CMSProductListComponent;
  let fixture: ComponentFixture<CMSProductListComponent>;
  let element: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        CMSProductListComponent,
        MockComponent({ selector: 'ish-product-tile-container', template: 'Product Tile', inputs: ['productSku'] }),
      ],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CMSProductListComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
    const pagelet = {
      definitionQualifiedName: 'fq',
      id: 'id',
      configurationParameters: {
        Products: ['1@Domain', '2@Domain'],
        Title: 'PageletTitle',
        CSSClass: 'css-class',
        ListItemCSSClass: 'li-css-class',
      },
    };
    component.pagelet = createSimplePageletView(pagelet);
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => component.ngOnChanges()).not.toThrow();
    expect(() => fixture.detectChanges()).not.toThrow();
    expect(element).toMatchSnapshot();
  });
});
