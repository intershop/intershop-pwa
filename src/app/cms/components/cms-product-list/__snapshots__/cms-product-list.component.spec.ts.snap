// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cms Product List Component should be created 1`] = `
<div
  id="root0"
  ng-version="7.1.4"
>
  
  <div
    class="product-list-container css-class"
    ng-reflect-klass="product-list-container"
    ng-reflect-ng-class="css-class"
  >
    
    <h2>
      PageletTitle
    </h2>
    <div
      class="product-list row"
    >
      
      <div
        class="product-list-item li-css-class"
        ng-reflect-klass="product-list-item"
        ng-reflect-ng-class="li-css-class"
      >
        <ish-product-tile-container
          ng-reflect-product-sku="1"
        >
          Product Tile
        </ish-product-tile-container>
      </div>
      <div
        class="product-list-item li-css-class"
        ng-reflect-klass="product-list-item"
        ng-reflect-ng-class="li-css-class"
      >
        <ish-product-tile-container
          ng-reflect-product-sku="2"
        >
          Product Tile
        </ish-product-tile-container>
      </div>
    </div>
  </div>
</div>
`;
